js-api-spec/value/list.test.ts

```diff
@@ -0,0 +1,283 @@
+                new SassString('b', {quotes: false}),
+                new SassString('c', {quotes: false}),
+              ],
+              {separator: ' '}
+            )
+          )
+        ).toBe(false);
+
+        expect(
+          list.equals(
+            new SassList(
+              [
+                new SassString('a', {quotes: false}),
+                new SassString('b', {quotes: false}),
+                new SassString('c', {quotes: false}),
+              ],
+              {separator: ',', brackets: true}
+            )
+          )
+        ).toBe(false);
+      });
+
+      it("doesn't equal a value with different contents", () => {
+        expect(
+          list.equals(
+            new SassList(
+              [
+                new SassString('a', {quotes: false}),
+                new SassString('x', {quotes: false}),
+                new SassString('c', {quotes: false}),
+              ],
+              {separator: ','}
+            )
+          )
+        ).toBe(false);
+      });
+    });
+
+    describe('Sass to JS index conversion', () => {
+      const list = new SassList([
+        new SassString('a'),
+        new SassString('b'),
+        new SassString('c'),
+      ]);
+
+      it('converts a positive index', () => {
+        expect(list.sassIndexToListIndex(new SassNumber(1))).toBe(0);
+        expect(list.sassIndexToListIndex(new SassNumber(2))).toBe(1);
+        expect(list.sassIndexToListIndex(new SassNumber(3))).toBe(2);
+      });
+
+      it('converts a negative index', () => {
+        expect(list.sassIndexToListIndex(new SassNumber(-1))).toBe(2);
+        expect(list.sassIndexToListIndex(new SassNumber(-2))).toBe(1);
+        expect(list.sassIndexToListIndex(new SassNumber(-3))).toBe(0);
+      });
+
+      it('rejects a non-number', () => {
+        expect(() =>
+          list.sassIndexToListIndex(new SassString('foo', {quotes: false}))
+        ).toThrow();
+      });
+
+      it('rejects a non-integer', () => {
+        expect(() => list.sassIndexToListIndex(new SassNumber(1.1))).toThrow();
+      });
+
+      it('rejects invalid indices', () => {
+        expect(() => list.sassIndexToListIndex(new SassNumber(0))).toThrow();
+        expect(() => list.sassIndexToListIndex(new SassNumber(4))).toThrow();
+        expect(() => list.sassIndexToListIndex(new SassNumber(-4))).toThrow();
+      });
+    });
+
+    describe('delimiters', () => {
+      it('defaults to comma separator and no brackets', () => {
+        const list = new SassList([
+          new SassString('a'),
+          new SassString('b'),
+          new SassString('c'),
+        ]);
+        expect(list.separator).toBe(',');
+        expect(list.hasBrackets).toBe(false);
+      });
+
+      it('supports null separators', () => {
+        const list = new SassList(
+          [new SassString('a'), new SassString('b'), new SassString('c')],
+          {separator: null}
+        );
+        expect(list.separator).toBe(null);
+      });
```

Comment:
A list with more than one element shouldn't allow an undecided separator. If that's not in the protocol/spec yet, it should be.

Reply:
Good point. I updated the protocol here: https://github.com/sass/embedded-protocol/pull/81. For the spec, would it be easier to update it here https://github.com/sass/sass/pull/3183?